cpp/dolfinx/geometry/utils.h

```diff
@@ -663,40 +663,36 @@ graph::AdjacencyList<std::int32_t> compute_colliding_cells(
 /// @param[in] mesh The mesh
 /// @param[in] points Points to check for collision (`shape=(num_points,
 /// 3)`). Storage is row-major.
+/// @param[in] cells Cells to check for ownership
```

Comment:
What is the sentinel value? i.e. What to pass if we want to check all cells? (`None` in Python).

Again, this could be converted to std::optional.

Reply:
I thought it was intentional to have optional arguments on the python side that are not optional on the c++ side. I copied this from an interpolate function:

https://github.com/FEniCS/dolfinx/blob/58f13df3668d03d4ba8d79ebb931e36880672950/python/dolfinx/fem/function.py#L412-L417
https://github.com/FEniCS/dolfinx/blob/58f13df3668d03d4ba8d79ebb931e36880672950/python/dolfinx/wrappers/fem.cpp#L416
https://github.com/FEniCS/dolfinx/blob/58f13df3668d03d4ba8d79ebb931e36880672950/cpp/dolfinx/fem/Function.h#L271-L273

For this interpolate function, `cells0` is optional on the Python side but not on the c++ side. Moreover `cells1` is optional in Python, non-optional in the wrapper and optional on the c++ side.

I'll commit a version where `cells` is optional. I won't touch other places of the code.

Reply:
I see this more of a hangover from not using std::optional in C++.

In any case I think we will do the full conversion to std::optional for `v0.10.0`.